src/library/dao/cuedao.cpp

```diff
@@ -48,9 +48,19 @@ CuePointer cueFromRow(const QSqlRecord& row) {
     VERIFY_OR_DEBUG_ASSERT(color) {
         return CuePointer();
     }
+    if (type == mixxx::CueType::Loop && length == 0) {
+        // These entries are likely added via issue #11283
+        qWarning() << "Discard loop cue" << hotcue << "found in database with length of 0";
+        return CuePointer();
+    }
+    if (type == mixxx::CueType::HotCue && position == 0 && *color == mixxx::RgbColor(0)) {
+        // These entries are likely added via issue #11283
+        qWarning() << "Discard black hot cue" << hotcue << "found in database at position 0";
+        return CuePointer();
+    }
```

Comment:
is there no better way to remove these invalid cuepoints from the db? maybe some sort of sanity check at startup or db migration?

Reply:
I was currently not able to re-add this issue. Maybe it is a pending 2.4 bug (keep investigating). 
If this is a case a library migration is not possible. 

A startup code will involve a startup penalty, because we need to check the whole library. So I think this on-the-fly removal is already a good option.

It could be an issue if we keep removing real cues that fit to the pattern.

Reply:
> A startup code will involve a startup penalty, because we need to check the whole library. So I think this on-the-fly removal is already a good option.

My issue is that this will only apply to tracks once they're loaded, in huge libraries, tracks with the deficiency could linger around for years until they've been loaded again. So this cleanup workaround will also have to stay forever. With a DB migration or something similar, the workaround could be removed after a couple versions when we can be certain that people that introduced this issue into their DB while testing the beta have cleaned it up by running a proper 2.4 release version once.

Reply:
Maybe this concern is overengineering, just sharing my thoughts. I'm not suggesting another approach would be better.

Reply:
Well I think both types of cues are reasonable to be removed, my complaint is just that it happens lazily on track while I would prefer a one-time cleanup action. 

In any case, we should definitely try to find the source of the garbage data. I'm not sure how to do that. A sanity check after the export could be an option, but that has some overhead associated with it. Do you have a better idea?

Reply:
I cannot tell yet. I need to investigate, maybe we should hold this PR back until we know more. 

The lazy clean up is simple and works without any other conditions. The only drawback I see is that we may have an issue if a user uses a version without this clean up code or a third party tool reading the Mixxx database. Not sure if that rectifies a more complex solution.

Reply:
I'd be in favor of finding the cause before trying to fix its symptoms.